source/cuda/cuda_matrix_free_smoother.cu

```diff
@@ -0,0 +1,148 @@
+/**************************************************************************
+ * Copyright (c) 2017-2019 by the mfmg authors                            *
+ * All rights reserved.                                                   *
+ *                                                                        *
+ * This file is part of the mfmg library. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the  *
+ * top-level directory                                                    *
+ *                                                                        *
+ * SPDX-License-Identifier: BSD-3-Clause                                  *
+ *************************************************************************/
+
+#include <mfmg/cuda/cuda_matrix_free_operator.cuh>
+#include <mfmg/cuda/cuda_matrix_free_smoother.cuh>
+#include <mfmg/cuda/cuda_matrix_operator.cuh>
+#include <mfmg/cuda/dealii_operator_device_helpers.cuh>
+#include <mfmg/cuda/utils.cuh>
+
+namespace mfmg
+{
+namespace
+{
+template <typename VectorType>
+struct SmootherOperator
```

Comment:
So all this is just for being able to specialize for `VectorType`. Do we actually need that if only allow for one type?